### ISSUE TITLE
No sign of Citation interface after opening README.Rmd template and similar files in visual mode

### ISSUE THREAD

**Issue by Robinlovelace** [rstudio/rstudio#11198]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.06.0-daily-396-amd64
    OS Version      : Ubuntu 20.04
    R Version       : 4.2.0

### Steps to reproduce the problem

Try adding citations either by entering `@` when in visual editor mode or with the insert below:

![image](https://user-images.githubusercontent.com/1825120/168180299-c5d5374b-b479-4cbc-97e8-d54518c9642c.png)

### Describe the problem in detail

There is no evidence of the citations interface that I know (knew) and love (loved)!

The raw text that appears after trying is like this: `\@test_2022`

### Describe the behavior you expected

I should be able to add citations.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

A bit more context: I have a large Zotero database and {rbbt} installed.

**Comment by Robinlovelace** [comment]
Interestingly I get citations when creating a new Quarto project. Will try creating a new R project now, guess it may be due to previous session info from previous project sessions.

![image](https://user-images.githubusercontent.com/1825120/168180738-dfeb8fcd-2b07-4b10-984b-5a547f4be004.png)

**Comment by Robinlovelace** [comment]
Update: I tried removing all files in `.Rproj.user` and re-opening the project. That didn't fix it.

**Comment by Robinlovelace** [comment]
There is definitely some weird inconsistent behaviour going on here. I create a new .Rmd document and citations pop-up fine. But when editing the file in which the citations failed previously there seems to be no way to add citations. Reporting here in case of use/interest.

**Comment by Robinlovelace** [comment]
Heads-up @ronblum I have just managed to reproduce this in the following environment:

```
RStudio Workbench
2022.06.0 Build 431.pro2
© 2009-2022 RStudio, PBC
"Spotted Wakerobin" Daily (4bc37c6a, 2022-05-20) for Ubuntu Bionic
Mozilla/5.0 (X11; Linux x86_64; rv:101.0) Gecko/20100101 Firefox/101.0
```

Reprex in following comment.

**Comment by Robinlovelace** [comment]
1. Create a new package project with the command `usethis::create_package("testpkg")`
2. Open a new .Rmd file, e.g. with the command `usethis::use_readme_rmd()` inside the new project
3. Start editing the RMarkdown file and try to add references to check if this issue is reproducible on other installations

**Comment by Robinlovelace** [comment]
https://user-images.githubusercontent.com/1825120/169608889-5542d28e-5e34-4ed7-880b-4dcdc1c823b6.mp4

Illustration of this.

**Comment by Robinlovelace** [comment]
Update: other .Rmd files in the same project have the same issue.

**Comment by Robinlovelace** [comment]
When creating a new project the citation options are there in a totally blank .Rmd doc:

![image](https://user-images.githubusercontent.com/1825120/169609414-13111a5d-5a77-4066-87ff-879e8d095de0.png)

**Comment by Robinlovelace** [comment]
OK progress: exactly the same file and project prevent citations being added when the contents of README.Rmd are overwritten by `usethis::use_readme_rmd()`.

**Comment by Robinlovelace** [comment]
Bingo, Pretty sure I've found the cause of this bug. Either the comments or the code chunk in the first 20 lines of this file mess with the citations dialogue making it fail for some reason:

```

---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# test2





The goal of test2 is to ...

```

**Comment by Robinlovelace** [comment]
Immediate question that would mitigate this for me and likely other users: how to revive the citation dialogue in projects that have suffered this fate?

**Comment by Robinlovelace** [comment]
Another discovery, this unwanted alert pops up sometimes. If you press it the citation dialogues seem to vanish as I've discovered to my cost (the little x is the one to press here):

![image](https://user-images.githubusercontent.com/1825120/169619171-df6d4616-fd72-445a-957b-475cad0cea6e.png)

**Comment by ronblum** [comment]
I'm able to reproduce this consistently in
- RStudio Workbench 2022.02.2+485.pro2
- RStudio Server 2022.06.0-daily+420

but not in RStudio Desktop 2022.06.0-daily+421. 

To reproduce:
1. Open a new R Markdown file. It's best to create an empty one, but not necessary.
2. Switch to source mode.
3. Empty the file of any contents, if it's not already empty.
4. Use @Robinlovelace's code above, at https://github.com/rstudio/rstudio/issues/11198#issuecomment-1133370664
5. Save the file.
6. Switch to visual mode.
7. Try to insert a citation.

Step 5 is important, because the problem doesn't seem to occur if you don't save before switching to visual mode.

**Comment by Robinlovelace** [comment]
[travis-main.zip](https://github.com/rstudio/rstudio/files/8755903/travis-main.zip)
 
No problem. See attached another maybe reproducible example. Every time I open this up in visual mode all the citations get changed from something like this:

```
[@wood_visualizing_2011; @vonschmidt_webbased_2019; @morgan2020; @beecham2022]
```

to this:

```
 [\@wood_visualizing_2011; \@vonschmidt_webbased_2019; \@morgan2020; \@beecham2022].
```

which is frustrating to say the least...

**Comment by Robinlovelace** [comment]
> It looks like a resurgence of #10075, at least the later variations—does this seem right?

Yes the resurgence of that issue seems to be a symptom of this bug.

**Comment by ronblum** [comment]
Created #11279 for the second issue (which is related to #10075).

**Comment by jjallaire** [comment]
There was an issue where there are no citations available in README files. This has been addressed here: https://github.com/rstudio/rstudio/commit/36db3473055f32d467f2e6f39028f45c9214c78a

**Comment by ronblum** [comment]
FWIW, this now works in RStudio Desktop 2022.06.0-daily+528 on MacOS 12.5.

I'm marking this as verified, but @Robinlovelace it would be great let us know if it works for you, as well!

**Comment by Robinlovelace** [comment]
Good news: this edge-case but important bug fixed in the latest daily version. Great job. A similarly important and annoying bug that is *not* yet fixed is #11313. I will comment on that as I see it as a major regression that for sure is reproducible and will bite lots of people.

**Comment by ronblum** [comment]
@Robinlovelace Thank you for checking! Glad to hear it's working. I've also updated the other issue, and reached out internally for some more help. ("I don't know what I'm doing" is implied, so I won't say so outright.)
